### ISSUE TITLE
What if the training dataset is too large to fit into memory?

### ISSUE THREAD

**Issue by paipai880429** [dnouri/nolearn#48]
@dnouri 
@cancan101 
Hey, guys. Thanks for sharing your ideas and bring endless contribution to nolearn community. I have a simple question when I am using nolearn.lasagne.NeuralNet() to do kaggle competition. After we construct a neural network architecture by calling NeuralNet() function, we then train the network by function network.fit(X_train, y_train). For small datasets such as MNIST, this is O.K. However, when the data size is too large so that it can not be loaded into memory one time, what should we do? Thank you for your tips.

**Comment by dnouri** [comment]
@paipai880429 One way is to call `fit()` multiple times.  Another one is to only put ids (or filenames) into `X` and then load the actual data based on these ids from disk in your batch iterator.

**Comment by paipai880429** [comment]
@dnouri Thank you for your timely reply. It seems that the former is more straightforward. By the way,  the document for nolearn in the index page(https://pythonhosted.org/nolearn/) seems not a detailed and latest version. Will any plans for a document update ?

**Comment by dnouri** [comment]
Yes, you're right.  nolearn.lasagne sadly has no documentation yet.  I'll be hopefully working on that soon.

**Comment by paipai880429** [comment]
@dnouri Dainel, would you please  explain ```Another one is to only put ids (or filenames) into X and then load the actual data based on these ids from disk in your batch iterator.``` in details? Are you suggesting overriding ```BatchIterator``` parameterized with ````file ID list``?

**Comment by rachidasen** [comment]
@dnouri if we call fit() multiple times won't it recreate validation set 20%, I am working on a project and I have divided the dataset into chunks of say 15k images and a total of 19 chunks, I have a separate validation set of say 17k images, Now if I call **.fit()** multiple times, it is creating a separate validation set each time 20% and concept of  epoch is also wrong as the term epoch is used for the whole training_set, Please clarify on this
